apis/management.cattle.io/v3/alerting_types.go

```diff
@@ -200,12 +200,10 @@ type CommonRuleField struct {
 
 type MetricRule struct {
 	Expression     string  `json:"expression,omitempty" norman:"required"`
-	LegendFormat   string  `json:"legendFormat,omitempty"`
-	Step           int64   `json:"step,omitempty"`
 	Description    string  `json:"description,omitempty"`
-	Duration       string  `json:"duration,omitempty"`
-	Comparison     string  `json:"comparison,omitempty" norman:"type=enum,options=equal|not-equal|greater-than|less-than|greater-or-equal|less-or-equal,default=equal"`
-	ThresholdValue float64 `json:"thresholdValue,omitempty" norman:"type=float"`
+	Duration       string  `json:"duration,omitempty" norman:"required"`
+	Comparison     string  `json:"comparison,omitempty" norman:"required,type=enum,options=equal|not-equal|greater-than|less-than|greater-or-equal|less-or-equal,default=equal"`
+	ThresholdValue float64 `json:"thresholdValue,omitempty" norman:"required,type=float"`
```

Comment:
do you think we can specify the default value for this parameter?

Reply:
This value depends on the expression, like is the express etcd leader not equal 1, the threshold is 1, the memory usage close to 80%, the thresholds is 0.8, so it hard to define a default value for it.